src/main/java/uk/gov/hmcts/cmc/claimstore/services/DocumentsService.java

```diff
@@ -0,0 +1,72 @@
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import java.util.Optional;
+
+@Service
+public class DocumentsService {
+    private final ClaimService claimService;
+    private final DefendantResponseCopyService defendantResponseCopyService;
+    private final LegalSealedClaimPdfService legalSealedClaimPdfService;
+    private final CountyCourtJudgmentPdfService countyCourtJudgmentPdfService;
+    private final DocumentManagementService documentManagementService;
+    private final DocumentManagementSealedClaimHandler documentManagementSealedClaimHandler;
+    private final boolean documentManagementFeatureEnabled;
+
+    @Autowired
+    public DocumentsService(
+        final ClaimService claimService,
+        final DefendantResponseCopyService defendantResponseCopyService,
+        final LegalSealedClaimPdfService legalSealedClaimPdfService,
+        final CountyCourtJudgmentPdfService countyCourtJudgmentPdfService,
+        final DocumentManagementService documentManagementService,
+        final DocumentManagementSealedClaimHandler documentManagementSealedClaimHandler,
+        @Value("${feature_toggles.document_management}") final boolean documentManagementFeatureEnabled
+    ) {
+        this.claimService = claimService;
+        this.defendantResponseCopyService = defendantResponseCopyService;
+        this.legalSealedClaimPdfService = legalSealedClaimPdfService;
+        this.countyCourtJudgmentPdfService = countyCourtJudgmentPdfService;
+        this.documentManagementService = documentManagementService;
+        this.documentManagementSealedClaimHandler = documentManagementSealedClaimHandler;
+        this.documentManagementFeatureEnabled = documentManagementFeatureEnabled;
+    }
+
+    public byte[] generateDefendantResponseCopy(final String claimExternalId) {
+        final Claim claim = claimService.getClaimByExternalId(claimExternalId);
+        return defendantResponseCopyService.createPdf(claim);
+    }
+
+    public byte[] generateLegalSealedClaim(final String claimExternalId, final String authorisation) {
+        final Claim claim = claimService.getClaimByExternalId(claimExternalId);
+        return downloadOrUploadAndGenerateLegalN1FormPdfDocument(claim, authorisation);
+    }
+
+    private byte[] downloadOrUploadAndGenerateLegalN1FormPdfDocument(final Claim claim, final String authorisation) {
+        final Optional<String> n1FormDocumentManagementPath = claim.getSealedClaimDocumentManagementSelfPath();
+
+        if (documentManagementFeatureEnabled && n1FormDocumentManagementPath.isPresent()) {
+            return documentManagementService.downloadDocument(authorisation, n1FormDocumentManagementPath.get());
+        } else {
+            documentManagementSealedClaimHandler.uploadRepresentativeSealedClaimToEvidenceStore(
+                new RepresentedClaimIssuedEvent(claim, Optional.empty(), authorisation)
+            );
+
+            return legalSealedClaimPdfService.createPdf(claim);
```

Comment:
this doesn't look right, multiple calls to the PDF service are being made

Reply:
rewriting call